archive/100-theorems-list/93_birthday_problem.lean

```diff
@@ -16,12 +17,86 @@ in terms of injective functions. The general result about `fintype.card (α ↪
 uses is `fintype.card_embedding_eq`.
 -/
 
-local notation `‖` x `‖` := fintype.card x
+local notation (name := finset.card)  `|` x `|` := finset.card x
+local notation (name := fintype.card) `‖` x `‖` := fintype.card x
 
-/-- **Birthday Problem** -/
+/-- **Birthday Problem**: set cardinality interpretation. -/
 theorem birthday :
   2 * ‖fin 23 ↪ fin 365‖ < ‖fin 23 → fin 365‖ ∧ 2 * ‖fin 22 ↪ fin 365‖ > ‖fin 22 → fin 365‖ :=
 begin
   simp only [nat.desc_factorial, fintype.card_fin, fintype.card_embedding_eq, fintype.card_fun],
   norm_num
 end
+
+section measure_theory
+
+open measure_theory
+open_locale probability_theory ennreal
+
+variables {n m : ℕ}
+
+/- In order for Lean to understand that we can take probabilities in `fin 23 → fin 365`, we must
+tell Lean that there is a `measurable_space` structure on the space. Note that this instance
+is only for `fin m` - Lean automatically figures out that the function space `fin n → fin m`
+is _also_ measurable, by using `measurable_space.pi` -/
+
+instance : measurable_space (fin m) :=
+{ measurable_set' := λ _, true,
+  measurable_set_empty := trivial,
+  measurable_set_compl := λ _ _, trivial,
+  measurable_set_Union := λ _ _, trivial }
+
+/- We then endow the space with a canonical measure, which is called ℙ. We define this to be
+the counting measure, scaled by the size of the whole set. -/
+noncomputable instance : measure_space (fin n → fin m) :=
+  ⟨(measure.count (set.univ : set $ fin n → fin m))⁻¹ • measure.count⟩
```

Comment:
Can you phrase the measure using `probability_theory.cond_count` (taking `s = set.univ`) here?